lib/beaker/dsl/helpers/host_helpers.rb

```diff
@@ -254,15 +254,29 @@ def rsync_to host, from_path, to_path, opts = {}
         # @option [String] archive_root The local directory to store the copied
         #                               file under. Defaults to
         #                               'archive/sut-files'.
+        # @option [String] archive_name The name of the archive to be copied to
+        #                               archive_root. Defaults to
+        #                               'sut-files.tgz'.
         #
         # @return [Result] Returns the result of the #scp_from operation.
-        def archive_file_from(host, from_path, opts = {}, archive_root = 'archive/sut-files')
+        def archive_file_from(host, from_path, opts = {}, archive_root = 'archive/sut-files', archive_name = 'sut-files.tgz')
+          require 'archive/tar/minitar'
           filedir = File.dirname(from_path)
           targetdir = File.join(archive_root, host.hostname, filedir)
           FileUtils.mkdir_p(targetdir)
-          scp_from(host, from_path, targetdir, opts)
+          result = scp_from(host, from_path, targetdir, opts)
+          return if result.exit_code != 0
```

Comment:
@smcelmurry so if the scp call fails, the circumstance here is that a user calls `archive_file_from`, and everything returns as if nothing went wrong (besides any reporting that `scp_from` might do itself), even though a failure happened?

It seems to me that we'll have a failure but won't return that that's the case, with the user only discovering this when they go to look at their archived files that don't exist. Is this not the case?

It also seems that `create_tarball` would be more resilient if it checked for file existence itself and failed if it was asked to tar a file that doesn't exist. Do you not agree, or is that being handled well by this code already that I'm not seeing?